django/university/views.py

```diff
@@ -198,6 +198,52 @@ def login(request):
     except requests.exceptions.RequestException as e:
         return JsonResponse({"error": e}, safe=False)
 
+"""
+    Returns schedule of student
+"""
+@api_view(["GET"])
+def student_schedule(request, student):
+    semana_ini = "20240101"
+    semana_fim = "20240601"
+
+    try:
+        url = f"https://sigarra.up.pt/feup/pt/mob_hor_geral.estudante?pv_codigo={student}&pv_semana_ini={semana_ini}&pv_semana_fim={semana_fim}" 
+        response = requests.get(url, cookies=request.COOKIES)
+
+        schedule_data = response.json()['horario']
+
+        for schedule in schedule_data:
+            schedule['ucurr_nome'] = course_unit_name(schedule['ocorrencia_id'])
+        
+        new_response = JsonResponse(schedule_data, safe=False)    
+
+        new_response.status_code = response.status_code
+
+        return new_response 
+    except requests.exceptions.RequestException as e:
+        return JsonResponse({"error": e}, safe=False)
+
+"""
+    Returns all classes of a course unit from sigarra
+""" 
+@api_view(["GET"])
+def schedule_sigarra(request, course_unit_id):
+    semana_ini = "20240101"
+    semana_fim = "20240601"
+
+    try:
+        url = f"https://sigarra.up.pt/feup/pt/mob_hor_geral.ucurr?pv_ocorrencia_id={course_unit_id}&pv_semana_ini={semana_ini}&pv_semana_fim={semana_fim}"
+        response = requests.get(url, cookies=request.COOKIES)
+
+        new_response = JsonResponse(response.json()['horario'], safe=False)
```

Comment:
This creates an exception and the response returns 500 to the client, not allowing the logout mechanism on the frontend to work correctly.